pkg/api/routes.go

```diff
@@ -140,7 +140,7 @@ func (rh *RouteHandler) CheckVersionSupport(response http.ResponseWriter, reques
 	response.Header().Set(constants.DistAPIVersion, "registry/2.0")
 	// NOTE: compatibility workaround - return this header in "allowed-read" mode to allow for clients to
 	// work correctly
-	if rh.c.Config.HTTP.AllowReadAccess {
```

Comment:
we do return the auth header in the authn.go handler, always.

that was an extra check, because if allowReadAccess was true and the request was a HEAD or a GET then the request would pass authn.go, and went into routes.go, and in the case of /v2/ which is checked by all tools, we needed to return the WWW-Authenticate header.

in other words without this check we wouldn't return the WWW-Authenticate  header for GET and HEAD request when allowReadAccess was true.
 
hope this is clear

Reply:
now, because of our change with anonymous policy, we may return a 403 with WWW-Authenticate headers, I wonder if tools know how to handle this.

Reply:
Actually it is not clear. I'm not saying we should remove the content of the if block. I'm saying we should always add the WWW-Authenticate header regardless of checkAnonymousPolicyExists. This just looks like an extra check which was redundant before and is not needed now either. Am I missing something?